File: gen_scripts/php/f21.loop.php
#!/usr/bin/env php
<?php
$answer = "";
// conditional loop with while
while  ($answer != "quit") {
    echo "Enter your name (quit to Exit): ";  # output prompt
    $answer = rtrim(fgets(STDIN));            # get input, trim newline
    if ($answer != "quit")
        echo "Hello $answer!\n";              # output result if not exiting
}
?>

File: gen_scripts/ruby/f21.loop.rb
#!/usr/bin/env ruby
answer = ""                               # default
# coditional loop using while/do...end construct
while answer != "quit" do
  print "Enter your name (quit to Exit): " # print prompt
  STDOUT.flush                            # flush buffer to show prompt
  answer = gets.chomp                     # get string input
 
  if answer != "quit"
      puts "Hello #{answer}!"             # print if not exiting
  end
end